docs/source/dev_guide/project-standards.rst

```diff
@@ -0,0 +1,205 @@
+
+Why squash and rebase commits?
+
+- Ensures build passes from the commit
+- Cleans up Git history for easy navigation
+- Makes collaboration and review process more efficient
+- Makes handling conflicts from rebasing simple since there are fewer commits to fix conflicts on
+- Makes ``git bisect`` easier and more effective to use. For example, it will show the exact commit that introduced a bug since the commit contains a relatively small changeset. On the otherhand, a merge commit bringing in many commits makes it much harder since it includes a large changeset. A changeset is a collection of commits that are logically grouped together.
+
+
+How to squash and rebase commits
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Assuming that you followed :ref:`"(b) Development Environment" <dev-env>`:
+
+1. Sync ``main`` with the main repo's ``main`` ::
+
+    git checkout main
+    git rebase <upstream-origin>/main
+    git push -f <fork-origin> main
+
+2. Get the SHA of the commit OR number of commits to rebase to ::
+
+    git log --graph --decorate --pretty=oneline --abbrev-commit
+
+3. Squash commits::
+
+    git rebase -i [SHA]
+
+    # OR
+
+    git rebase -i HEAD~[NUMBER OF COMMITS]
+
+4. Make sure your squashed commit messages are refined
+
+5. Rebase branch onto ``main`` ::
+
+    git checkout <branch-name>
+    git rebase main
+    git push -f <fork-origin> <branch-name>
+
+6. Force push to remote branch ::
+
+    # You have to force push because the rebase rewrites the commit SHAs
+    git push -f <fork-origin> <branch-name>
+
+Source:
+https://blog.carbonfive.com/always-squash-and-rebase-your-git-commits/
+
+Code Quality Assurance
+----------------------
+
+This project uses several tools for code formatting, linting, and type checking listed below.
+
+- Code Formatting: `black <https://black.readthedocs.io/en/stable/>`__
+- Linting: `flake8 <https://github.com/PyCQA/flake8#flake8>`__, `isort <https://pycqa.github.io/isort/>`__
+- Optional Type Checking: `mypy <http://mypy-lang.org/>`__
+
+You can run them as hooks manually/automatically when committing using ``pre-commit``, or manually through the terminal or IDE/text editor.
+
+Helpful Commands
+~~~~~~~~~~~~~~~~
+
+Run a tool
+    ::
+
+       # Available tool names: black, flake8, isort, mypy
+       <tool_name> .
+
+.. _ci-cd:
+
+Continuous Integration / Continuous Delivery (CI/CD)
+----------------------------------------------------
+
+This project uses `GitHub Actions <https://github.com/E3SM-Project/zppy/actions>`_ to run two CI/CD workflows.
+
+1. CI/CD Build Workflow
+
+  This workflow is triggered by Git ``pull_request`` and ``push`` (merging PRs) events to the the main repo's ``main``.
+
+  Jobs:
+
+    1. Run ``pre-commit`` for formatting, linting, and type checking
+    2. Run test suite in a conda environment
+    3. Publish latest `main` docs (depends on jobs 1 and 2)
```

Comment:
With `check-jobs-to-skip`, 3 should be:
```
3. Publish latest ``main`` docs (only on ``push``)
```
since it can occur in parallel with the other jobs.